crypto/pkcs11/pkcs11helpers.go

```diff
@@ -208,13 +208,14 @@ func pkcs11UriGetLoginParameters(p11uri *pkcs11uri.Pkcs11URI, privateKeyOperatio
 	return pin, module, slotid, nil
 }
 
-// pkcs11UriGetKeyLabel gets the key label by retrieving the value of the 'object' attribute
-func pkcs11UriGetKeyLabel(p11uri *pkcs11uri.Pkcs11URI) (string, error) {
-	serial, ok := p11uri.GetPathAttribute("object", false)
-	if !ok {
-		return "", errors.New("No 'object' attribute found in pkcs11 URI")
-	}
-	return serial, nil
+// pkcs11UriGetKeyIdAndLabel gets the key label by retrieving the value of the 'object' attribute
+func pkcs11UriGetKeyIdAndLabel(p11uri *pkcs11uri.Pkcs11URI) (string, string, error) {
+	keyid, ok2 := p11uri.GetPathAttribute("id", false)
+	label, ok1 := p11uri.GetPathAttribute("object", false)
+	if !ok1 && !ok2 {
+		return "", "", errors.New("Neither 'id' nor 'object' attributes were found in pkcs11 URI")
+	}
+	return keyid, label, nil
```

Comment:
clarification: so if keyid is empty its still okay?

Reply:
Yes, it would be ok. Both are not required.